crates/core/src/sync_local.rs

```diff
@@ -372,3 +398,175 @@ SELECT
+    fn put_statement(
+        &mut self,
+        db: *mut sqlite::sqlite3,
+    ) -> Result<&PreparedPendingStatement, SQLiteError> {
+        let cache_slot = &mut self.cached_put;
+        if let None = cache_slot {
+            let stmt = PreparedPendingStatement::prepare(db, &self.definition.put)?;
+            *cache_slot = Some(stmt);
+        }
+
+        return Ok(cache_slot.as_ref().unwrap());
+    }
+
+    fn delete_statement(
+        &mut self,
+        db: *mut sqlite::sqlite3,
+    ) -> Result<&PreparedPendingStatement, SQLiteError> {
+        let cache_slot = &mut self.cached_delete;
+        if let None = cache_slot {
+            let stmt = PreparedPendingStatement::prepare(db, &self.definition.delete)?;
+            *cache_slot = Some(stmt);
+        }
+
+        return Ok(cache_slot.as_ref().unwrap());
+    }
+}
+
+impl<'a> ParsedSchemaTable<'a> {
+    pub const fn json_table() -> Self {
+        Self { raw: None }
+    }
+
+    pub fn raw(definition: &'a RawTable) -> Self {
+        Self {
+            raw: Some(RawTableWithCachedStatements {
+                definition,
+                cached_put: None,
+                cached_delete: None,
+            }),
+        }
+    }
+}
+
+struct PreparedPendingStatement<'a> {
+    stmt: ManagedStmt,
+    params: &'a [PendingStatementValue],
+}
+
+impl<'a> PreparedPendingStatement<'a> {
+    pub fn prepare(
+        db: *mut sqlite::sqlite3,
+        pending: &'a PendingStatement,
+    ) -> Result<Self, ResultCode> {
+        let stmt = db.prepare_v2(&pending.sql)?;
+        // TODO: Compare number of variables / other validity checks?
+
+        Ok(Self {
+            stmt,
+            params: &pending.params,
+        })
+    }
+
+    pub fn bind_for_put(&self, id: &str, json_data: &serde_json::Value) -> Result<(), SQLiteError> {
+        use serde_json::Value;
+        for (i, source) in self.params.iter().enumerate() {
+            let i = (i + 1) as i32;
+
+            match source {
+                PendingStatementValue::Id => {
+                    self.stmt.bind_text(i, id, Destructor::STATIC)?;
+                }
+                PendingStatementValue::Column(column) => {
+                    let parsed = json_data.as_object().ok_or_else(|| {
+                        SQLiteError(
+                            ResultCode::CONSTRAINT_DATATYPE,
+                            Some("expected oplog data to be an object".to_string()),
+                        )
+                    })?;
+
+                    match parsed.get(column) {
+                        Some(Value::Bool(value)) => {
+                            self.stmt.bind_int(i, if *value { 1 } else { 0 })
+                        }
+                        Some(Value::Number(value)) => {
+                            if let Some(value) = value.as_f64() {
+                                // ??? there's no bind_double???
+                                self.stmt.bind_int64(i, value as i64)
```

Comment:
What will happen in this case - I assume the value is rounded down? Should we rather error or use null in this case for now, until we add support in `sqlite-rs-embedded`?

Reply:
Let's just do it right from the beginning: https://github.com/powersync-ja/sqlite-rs-embedded/pull/3